tests/jerry/es2015/regression-test-issue-2782.js

```diff
@@ -0,0 +1,21 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var arrayLikeObj = { "0" : 0, "1" : 1, "2" : 2, "length": 3 }
+
+iterator = Array.prototype.keys.call(arrayLikeObj);
+assert (JSON.stringify (iterator.next()) === '{"value":0,"done":false}')
```

Comment:
@rerobika  This JSON stringify comparison is not correct. the order of the properties is not defined for an object and thus the stringify can also have different results.

Reply:
Good catch! Fixed.